### ISSUE TITLE
Document Relationship of winpty-agent.exe with Alacritty on Windows

### ISSUE THREAD

**Issue by matttproud** [jwilm/alacritty#1818]
I realize you closed out initial Windows support not too long ago (well done).  I want to ask, it appears `winpty-agent.exe` is required to run Alacritty on Windows, but the mechanism for installation and use seems unclear.

Does `winpty-agent.exe` set it self up onto a Windows install automatically, or does one have to start it as a companion process on login?  Is it a one-time thing?  It'd be well to document this.

In any case, big thank you and the community for landing this support.  I cannot describe how nice it feels to use a spartan, yet modern terminal here.

**Comment by mariusmg** [comment]
>Does winpty-agent.exe set it self up onto a Windows install automatically, or does one have to start it as a companion process on login?

I just dumped winpty.dll and winpty-agent.exe in the same folder as alacritty and it worked fine.


> I cannot describe how nice it feels to use a spartan, yet modern terminal here.

Why not use ConEmu then ? :)

**Comment by matttproud** [comment]
I figured this out: the winpty library expects to find the agent executable in the same directory and only the same directory, no search path, environment, nothing.  This gave me a panic at runtime.  I'll produce a patch to handle this more elegantly.

> Why not use ConEmu then ? :)

ConEmu looks and feel super bloated in comparison.  To me, a terminal emulator is close to bare bones, like `st`, `alacritty`, or `xterm`.  I also appreciate the UNIX-like ergonomics with Alacritty's cut-and-paste.

**Comment by Jiehong** [comment]
Hi,

I just wanted to test this on Windows. I encountered the same panic, so I added the winpty.exe and dll file where the Alacritty executable is located, but when I try to launch it, I get another panic (and a short lived, white window for Alacritty).

Running it from the terminal gives the following output:

```
 RUST_BACKTRACE=1 ./Alacritty-v0.2.3.exe
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Err { ptr: winpty_error_s { _unused: [] }, code: 6, message: "agent died" }', libcore\result.rs:1009:5
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: winpty_spawn_config_new
  11: BaseThreadInitThunk
  12: RtlUserThreadStart
```

Looks weird.

**Comment by zacps** [comment]
@Jiehong The required file is `winpty-agent.exe`, `winpty.exe` is for a different purpose and no DLLs are required.

**Comment by Jiehong** [comment]
@zacps I made a typo in my comment, I actually added `winpty-agent.exe`, and not `winpty.exe`. I just checked to make sure. Still creating this panic though.

**Comment by jdhao** [comment]
I have downloaded Alacritty-v0.2.3.exe and winpty-agent.exe from the release page.  The two executable files are in the same directory. 
When I run Alacritty-v0.2.3.exe in the Windows cmd or cmder,  I get the same error message:

```
thread 'main' panicked at 'gl function was not loaded', C:\Users\travis\build\jwilm\alacritty\target\release\build\alacritty-9fada6ef0ea67072\out/gl_bindings.rs:20624:13
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

How do we run it correctly (which terminal, detailed command, set up, etc.)? There seems no documentation.

**Comment by zacps** [comment]
@jdhao Could you create another issue please? 

Please include what GPU you have as well.

**Comment by OmriSama** [comment]
is this being handled or could I take a stab at it with a pull request?

**Comment by zacps** [comment]
There's a linked PR already. I just need to test deployment. Waiting on motivation more than anything.
